src/main/java/com/sun/librarymanagement/domain/service/impl/BookServiceImpl.java

```diff
@@ -126,11 +127,16 @@ public PaginatedResponseDto<BookResponseDto> search(
     public BookResponseDto favorite(long id, Long currentUserId) {
         BookEntity currentBook = bookRepository.findByIdWithLock(id).orElseThrow(() -> new AppException(AppError.BOOK_NOT_FOUND));
         UserEntity userEntity = userService.getUserById(currentUserId);
-        boolean isFavorite = currentBook.getFavorites().stream().anyMatch((e) -> e.getId().equals(userEntity.getId()));
+        boolean isFavorite = currentBook.getFavorites() != null &&
+            currentBook.getFavorites().stream().anyMatch((e) -> e.getId().equals(userEntity.getId()));
```

Comment:
```suggestion
        boolean isFavorite = Optional.ofNullable(currentBook.getFavorites())
    .map(favorites -> favorites.stream().anyMatch(e -> e.getId().equals(userEntity.getId())))
    .orElse(false);
```
khuyến khích dùng Optional nhá em